packages/core/src/browser/about-dialog.tsx

```diff
@@ -0,0 +1,76 @@
+/********************************************************************************
+ * Copyright (C) 2018 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import * as React from 'react';
+import { inject, injectable, postConstruct } from 'inversify';
+import { ReactDialog, DialogProps } from './dialogs';
+import { ApplicationServer, ApplicationInfo, ExtensionInfo } from '../common/application-protocol';
+
+export const ABOUT_CONTENT_CLASS = 'theia-aboutDialog';
+export const ABOUT_EXTENSIONS_CLASS = 'theia-aboutExtensions';
+
+@injectable()
+export class AboutDialogProps extends DialogProps {
+}
+
+@injectable()
+export class AboutDialog extends ReactDialog {
+
+    protected readonly okButton: HTMLButtonElement;
+
+    @inject(ApplicationServer)
+    protected readonly appServer: ApplicationServer;
+
+    constructor(
+        @inject(AboutDialogProps) protected readonly props: AboutDialogProps
+    ) {
+        super({
+            title: props.title
+        });
+    }
+
+    protected applicationInfo: ApplicationInfo | undefined;
+    protected extensionsInfos: ExtensionInfo[] = [];
+
+    @postConstruct()
+    protected async init(): Promise<void> {
+        this.applicationInfo = await this.appServer.getApplicationInfo();
+        this.extensionsInfos = await this.appServer.getExtensionsInfos();
+        this.update();
+        this.appendAcceptButton('Ok');
+    }
+
+    protected render(): React.ReactNode {
+        const applicationInfo = this.applicationInfo;
+        const extensionsInfos = this.extensionsInfos;
+
+        return <div style={{ flex: '1 100%', paddingBottom: 'calc(var(--theia-ui-padding) * 3)' }} className={ABOUT_CONTENT_CLASS}>
+            {applicationInfo && <h3>{applicationInfo.name} {applicationInfo.version}</h3>}
+
+            <h3>List of extensions</h3>
+
+            <ul className={ABOUT_EXTENSIONS_CLASS}>
+                {
+                    extensionsInfos
+                        .sort((a: ExtensionInfo, b: ExtensionInfo) => a.name.toLowerCase().localeCompare(b.name.toLowerCase()))
+                        .map((extension: ExtensionInfo) => <li key={extension.name}>{extension.name} {extension.version}</li>)
+                }
+            </ul>
+        </div>;
+    }
```

Comment:
This is so much better as a React dialog.  A lot of the React rendering in Theia is done by breaking out each section and sub-section into different functions.  This makes it easy for extenders (developers creating custom Theia-based IDEs) to be-bind, extend, and re-implement just a single part of the render.  So here, split out perhaps the two <h3> sections to two further protected methods.  I know I have appreciated it when this was done elsewhere in Theia.

Reply:
Just to clarify, you do mean something like

```jsx
{this.renderHead()}
{this.renderExtensions()}
```

which I've seen done elsewhere, and not a React component, right? Although a React component would be nice, if you can somehow make them live as class methods somehow.

Reply:
Yes, exactly.  A React component would be overkill here.

Reply:
Not that overkill...

```diff
diff --git a/packages/core/src/browser/about-dialog.tsx b/packages/core/src/browser/about-dialog.tsx
index ec1b533e8..cc671ef8c 100644
--- a/packages/core/src/browser/about-dialog.tsx
+++ b/packages/core/src/browser/about-dialog.tsx
@@ -53,12 +53,12 @@ export class AboutDialog extends ReactDialog {
         this.appendAcceptButton('Ok');
     }
 
-    protected renderHeader(): React.ReactNode {
+    protected Header(): React.ReactElement {
         const applicationInfo = this.applicationInfo;
-        return applicationInfo && <h3>{applicationInfo.name} {applicationInfo.version}</h3>;
+        return <>{applicationInfo && <h3>{applicationInfo.name} {applicationInfo.version}</h3>}</>;
     }
 
-    protected renderExtensions(): React.ReactNode {
+    protected Extensions(): React.ReactElement {
         const extensionsInfos = this.extensionsInfos;
         return <>
             <h3>List of extensions</h3>
@@ -75,8 +75,8 @@ export class AboutDialog extends ReactDialog {
 
     protected render(): React.ReactNode {
         return <div style={{ flex: '1 100%', paddingBottom: 'calc(var(--theia-ui-padding) * 3)' }} className={ABOUT_CONTENT_CLASS}>
-            {this.renderHeader()}
-            {this.renderExtensions()}
+            <this.Header />
+            <this.Extensions />
         </div>;
     }
```

Reply:
OK, you are correct that form is not overkill.  In Theia the first form (in red above) seems to be the convention.  I thought you meant putting each into a separate class that extends React.Component (which is done in Theia code if it needs sharing across classes or needs to manage its own state).